electron/app/jobs/save-to-db.ts

```diff
@@ -49,12 +49,387 @@ export type user_events_insert_input = {
+                    throw new Error('Token Expired!');
+                }
+
+                const possibleEntitiesData = await fetchGraphQLClient(
+                    process.env.HASURA_GRAPHQL_ENGINE_DOMAIN,
+                    {
+                        token: this.token,
+                    },
+                )<PossibleEntities, { after: string }>({
+                    operationAction: `
+                        query fetchPossibleEntities($after: timestamptz!) {
+                            tasks(where: {
+                                status: {_nin: [finished, cancelled]}
+                                createdAt: {_gte: $after}
+                            }) {
+                                id
+                                ticketCode
+                                taskCode
+                            }
+                            tickets(where: {
+                                createdAt: {_gte: $after}
+                            }) {
+                                id
+                                code
+                            }
+                            projects: client_projects(where: {
+                                _or: [
+                                    {deletedAt: {_is_null: true}},
+                                    {deletedAt: {_gte: "now()"}}
+                                ]
+                            }) {
+                                id
+                                name
+                            }
+                            clients(where: {
+                                _or: [
+                                    {churnedAt: {_is_null: true}},
+                                    {churnedAt: {_gte: "now()"}},
+                                ]
+                            }) {
+                                id
+                                name
+                            }
+                        }
+                    `,
+                    variables: {
+                        after: moment().subtract(15, 'days').toJSON(),
+                    },
+                });
+
+                if ((possibleEntitiesData.errors?.length ?? 0) > 0) {
+                    throw new Error(
+                        `Error fetching possible entities from GitStart's DB | ${JSON.stringify(
+                            possibleEntitiesData.errors,
+                        )}`,
+                    );
+                }
+
+                const chunks = chunkEvents(events, {
+                    disableSorting: true,
+                    chunkEvery: 300, // chunks every 300 seconds (5 minutes)
+                });
+
+                const summary = chunks.reduce<SummaryItem[]>((summary, chunk) => {
+                    const summarizedEvents = summarizeEvents(chunk, {
+                        threshold: 0.3, // any activity that takes 30% or more of the chunk is considered a possible summarizer of the whole chunk.
```

Comment:
why not higher? or did this come from trial and error

Reply:
no we could bump it higher, but I thought 0.3 was a good starting point as lets say the user is working on GED-123 in VS Code and localhost:3000 for 7 minutes in total and 3 minutes of something else like "learning React", those 10 minutes gets split into two 5-minute chunks. The first chunk gets summarized as being GED-123 as the majority of time was spent on that, but in the second chunk, only 2/5th of the chunk is GED-123, so for GED-123 to be considered, then the threshold has to be set below 0.4 (=2/5). Then once it is considered the algorithm compares priority of a task vs a learning activity, in which it prefers tasks.